File: chatmessageview/src/main/java/com/github/bassaer/chatmessageview/utils/MessageDateComparator.java
package com.github.bassaer.chatmessageview.utils;

import com.github.bassaer.chatmessageview.models.Message;

import java.util.Comparator;

public class MessageDateComparator implements Comparator<Message> {
    @Override
    public int compare(Message first, Message second) {
        if (first.getCreatedAt().before(second.getCreatedAt())) {
            return -1;
        }
        if (first.getCreatedAt().after(second.getCreatedAt())) {
            return 1;
        }
        return 0;
    }
}

File: chatmessageview/src/main/kotlin/com/github/bassaer/chatmessageview/util/MessageDateComparator.kt
package com.github.bassaer.chatmessageview.util

import com.github.bassaer.chatmessageview.model.Message

import java.util.*

class MessageDateComparator : Comparator<Message> {
    override fun compare(first: Message, second: Message): Int {
        if (first.sendTime.before(second.sendTime)) {
            return -1
        }
        return if (first.sendTime.after(second.sendTime)) {
            1
        } else 0
    }
}
